lucene/core/src/java/org/apache/lucene/index/SortingCodecReader.java

```diff
@@ -1090,20 +1150,64 @@ public Bits getLiveDocs() {
   }
 
   @Override
-  public PointValues getPointValues(String fieldName) throws IOException {
-    final PointValues inPointValues = in.getPointValues(fieldName);
-    if (inPointValues == null) {
-      return null;
-    } else {
-      return new SortingPointValues(inPointValues, docMap);
-    }
+  public PointsReader getPointsReader() {
+    return new PointsReader() {
+      @Override
+      public void checkIntegrity() throws IOException {
+        in.getPointsReader().checkIntegrity();
+      }
+
+      @Override
+      public PointValues getValues(String field) throws IOException {
+        final PointValues inPointValues = in.getPointValues(field);
```

Comment:
IMO we should delegate to `in#getPointsReader()` as the contract is slightly different on the reader and the points reader (the reader returns null if the field was not indexed with points while the points reader fails).